Migration script to update the data layer for the additional sections

fatimarabab opened the pull request:
## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] My code follows the style guidelines enforced by static analysis tools.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.

<!-- Delete the tasks from the above list that are Not Applicable for your pull request -->

## Pull Request Details

Wrike Ticket- [Create a migration script to update the data layer for the additional sections in the analysis that are within presentations as of Spring 2023](https://www.wrike.com/open.htm?id=1112510444)
<!-- Note: Title your Pull Request with an appropriate title corresponding to the Issue title -->
Changes made:

-Wrote a migration script to update the analysis with the additional sections in the annotations. 

**To Review:**
<!-- Make a to do list of things to check for to approve the pull request -->
<!-- Modify the below list as appropriate by editing and deleting text that is not applicable-->

- [x] Static Analysis by Reviewer
- [x] Test if the migration script works correctly
     - Switch to the `migration-script-additional-sections` branch
     
    ````bash
      # From the root folder of Rosalution

       docker system prune -a --volumes
       docker-compose up --build -d
     ````

     - Navigate to `http://local.rosalution.cgds/rosalution` and login with `developer`
     - Click on the `CPAM0002` card and then the `VMA21` gene at the top of the AnalysisView
     - Scroll to any of the AnnotationSections titled `Modelability` and `Protein Expression` and attach an image
     - Run the migration script to rename the old annotation sections to the new additional annotation sections, with the following command
     
     ```bash
      docker compose exec rosalution-db  mongosh /tmp/fixtures/migrations/migrate-additional-sections.js
     ```
     - Check the database is updated, by confirming that the gene `VMA21` in `genomic_units` includes the annotations for `Orthology` and `Human Gene Expression` in its `annotations` object.
     
     ```bash
      # Activate mongo shell
      docker exec -it rosalution-rosalution-db-1  mongosh
      use rosalution_db                          
      db.genomic_units.find({'gene': 'VMA21'});
     ```
   - ***Without bringing down docker and while currently navigated to`CPAM0002` `VMA21` annotation page***, Git checkout to the `migration-script-add-sections` branch
   - Ensure that `Orthology` and `Human Gene Expression` have updated with the images under those sections.
- [x] All Github Actions checks have passed.

SeriousHorncat commented on etc/fixtures/migrations/migrate-additional-sections.js:
```diff
@@ -0,0 +1,55 @@
+
+const annotationSectionsRename = {
+  'Modelability': 'Orthology',
+  'Protein Expression': 'Human Gene Expression',
```
```suggestion
  'Protein Expression': 'Human_Gene_Expression',
```
The following target dataset name needs the underscores in it to match directly to what the configuration is looking for.

SeriousHorncat commented:
Confirmed multiple images attached also work with the migration.
<img width="805" alt="Screenshot 2023-06-07 at 2 16 04 PM" src="https://github.com/uab-cgds-worthey/rosalution/assets/1209059/72552c03-fbad-4817-b9fa-7251a4c4ca98">

fatimarabab commented:
Accepted and added the suggestion with the dataset name to match the configuration in target branch.
Also, switched out the target branch from main to https://github.com/uab-cgds-worthey/rosalution/tree/annotation-rendering-configuration-new-section-updates.
Thank you!

SeriousHorncat commented:
Confirmed changes work
<img width="600" alt="Screenshot 2023-06-07 at 2 32 04 PM" src="https://github.com/uab-cgds-worthey/rosalution/assets/1209059/37152cf8-8b26-4279-b355-75736eed1abc">

SeriousHorncat reviewed (approved):
Ready to merge when tests pass!
